via/views/view_pdf.py

```diff
@@ -23,81 +23,61 @@ def view_pdf(context, request):
+        "proxy_pdf_url": request.find_service(PDFURLBuilder).get_pdf_url(url),
         "client_embed_url": request.registry.settings["client_embed_url"],
         "static_url": request.static_url,
         "hypothesis_config": h_config,
     }
 
 
-class ProxyURLBuilder:
-    @staticmethod
-    def google_file_url(request, file_details, url):
-        route = "proxy_google_drive_file"
-        if file_details.get("resource_key"):
-            route += ":resource_key"
-
-        return request.find_service(SecureLinkService).sign_url(
-            request.route_url(
-                route,
-                # Pass the original URL along so it will show up nicely in
-                # error messages. This isn't useful for users as they don't
-                # see this directly, but it's handy for us.
-                _query={"url": url},
-                **file_details,
-            )
-        )
-
-    @staticmethod
-    def nginx_pdf_url(url, nginx_server, secret):
-        """Return the URL from which the PDF viewer should load the PDF."""
-
-        # Compute the expiry time to put into the URL.
-        exp = int(quantized_expiry(max_age=timedelta(hours=25)).timestamp())
-
-        # The expression to be hashed.
-        #
-        # This matches the hash expression that we tell the NGINX secure link
-        # module to use with the secure_link_md5 setting in our NGINX config file.
-        #
-        # http://nginx.org/en/docs/http/ngx_http_secure_link_module.html#secure_link_md5
-        hash_expression = f"/proxy/static/{exp}/{url} {secret}"
-
-        # Compute the hash value to put into the URL.
-        #
-        # This implements the NGINX secure link module's hashing algorithm:
-        #
-        # http://nginx.org/en/docs/http/ngx_http_secure_link_module.html#secure_link_md5
-        hash_ = hashlib.md5()
-        hash_.update(hash_expression.encode("utf-8"))
-        sec = hash_.digest()
-        sec = b64encode(sec)
-        sec = sec.replace(b"+", b"-")
-        sec = sec.replace(b"/", b"_")
-        sec = sec.replace(b"=", b"")
-        sec = sec.decode()
-
-        # Construct the URL, inserting sec and exp where our NGINX config file
-        # expects to find them.
-        return f"{nginx_server}/proxy/static/{sec}/{exp}/{url}"
+@view_config(route_name="proxy_google_drive_file", decorator=(has_secure_url_token,))
+@view_config(
+    route_name="proxy_google_drive_file:resource_key", decorator=(has_secure_url_token,)
+)
+def proxy_google_drive_file(request):
+    """Proxy a file from Google Drive."""
+
+    # Add an iterable to stream the content instead of holding it all in memory
+    content_iterable = request.find_service(GoogleDriveAPI).iter_file(
+        file_id=request.matchdict["file_id"],
+        resource_key=request.matchdict.get("resource_key"),
+    )
+
+    return _iter_pdf_response(request.response, content_iterable)
+
+
+def _iter_pdf_response(response, content_iterable):
+
+    try:
```

Comment:
```suggestion
    try:
```

Saving the whales one new line at a time. :whale: